### DISCUSSION TITLE
Prometheus support

### DISCUSSION THREAD

**Discussion by estahn** [aquasecurity/kube-bench#676] [Development]
I think it would be a good idea to continuously report on this benchmark and possibly have alerts being triggered to actively resolve these.

This could be done via prometheus and specifically by using the https://github.com/prometheus/pushgateway

Thoughts?

**Comment by lizrice** [comment]
This sounds like a nice idea, let me see if I am understanding this correctly. Whenever you run a kube-bench job it would report failures to the pushgateway, which aggregates the results for Prometheus to scrape at its leisure? Can you run one pushgateway to handle kube-bench results from multiple nodes?

**Reply by estahn** [reply]
@lizrice Yes, your understanding is correct. You can use a single pushgateway, it just requires to have a differentiator in the metric:

```
{node="foo1",...}
{node="foo2",...}
```
